pkgs/applications/misc/warpd/default.nix

```
@@ -2,48 +2,37 @@
 , stdenv
 , fetchFromGitHub
 , git
-, libXi
-, libXinerama
-, libXft
-, libXfixes
-, libXtst
-, libX11
-, libXext
-, waylandSupport ? false, cairo, libxkbcommon, wayland
+, withWayland ? true, cairo, libxkbcommon, wayland
+, withX ? true, libXi, libXinerama, libXft, libXfixes, libXtst, libX11, libXext
 }:
 
 stdenv.mkDerivation rec {
   pname = "warpd";
-  version = "1.3.4";
+  version = "1.3.5";
 
   src = fetchFromGitHub {
     owner = "rvaiya";
     repo = "warpd";
     rev = "v${version}";
-    sha256 = "sha256-aNv2/+tREvKlpTAsbvmFxkXzONNt73/061i4E3fPFBM=";
+    sha256 = "sha256-5B3Ec+R1vF2iI0ennYcsRlnFXJkSns0jVbyAWJA4lTU=";
     leaveDotGit = true;
   };
 
   nativeBuildInputs = [ git ];
 
-  buildInputs =  if waylandSupport then [
-    cairo
-    libxkbcommon
-    wayland
-  ] else [
-    libXi
-    libXinerama
-    libXft
-    libXfixes
-    libXtst
-    libX11
-    libXext
-  ];
+  buildInputs =
+    []
+    ++ lib.optionals withWayland
+      [ cairo libxkbcommon wayland ]
+    ++ lib.optionals withX
+      [ libXi libXinerama libXft libXfixes libXtst libX11 libXext ];
```

I think this will look much cleaner.
```suggestion
  buildInputs = lib.optionals withWayland [
    cairo
    libxkbcommon
    wayland
  ] ++ lib.optionals withX [
    libXi
    libXinerama
    libXft
    libXfixes
    libXtst
    libX11
    libXext
  ];
```